components/org.wso2.carbon.identity.api.server.fetch.remote/org.wso2.carbon.identity.api.server.fetch.remote.v1/src/main/java/org/wso2/carbon/identity/api/server/fetch/remote/v1/core/ServerRemoteFetchConfigManagementService.java

```diff
@@ -0,0 +1,814 @@
+            for (PushEventWebHookPOSTRequestCommits commit : commits) {
+                fileNames.addAll(commit.getAdded());
+                fileNames.addAll(commit.getModified());
+            }
+            return fileNames;
+        } else {
+            throw handleException(Response.Status.BAD_REQUEST, RemoteFetchConfigurationConstants.
+                    ErrorMessage.ERROR_CODE_COMMIT_NOT_FOUND, null);
+        }
+    }
+
+    /**
+     * This method is used to handle remote fetch core exception and create API error wit suitable response code and
+     * status by checking its instance type.
+     *
+     * @param e         RemoteFetchCoreException.
+     * @param errorEnum RemoteFetchConfigurationConstants.ErrorMessage
+     * @param data      data
+     * @return APIError
+     */
+    private APIError handleRemoteFetchConfigurationException(RemoteFetchCoreException e,
+                                                             RemoteFetchConfigurationConstants.ErrorMessage errorEnum,
+                                                             String data) {
+
+        ErrorResponse errorResponse = getErrorBuilder(errorEnum, data).build(log, e, errorEnum.getDescription());
+
+        Response.Status status;
+
+        if (e instanceof RemoteFetchClientException) {
+            if (e.getErrorCode() != null) {
+                String errorCode = e.getErrorCode();
+                errorCode =
+                        errorCode.contains(RemoteFetchConfigurationConstants.ERROR_CODE_DELIMITER) ?
+                                errorCode : RemoteFetchConfigurationConstants.
+                                REMOTE_FETCH_CONFIGURATION_MANAGEMENT_PREFIX + errorCode;
+                errorResponse.setCode(errorCode);
+            }
+            errorResponse.setDescription(e.getMessage());
+            status = Response.Status.BAD_REQUEST;
+        } else if (e instanceof RemoteFetchServerException) {
+            if (e.getErrorCode() != null) {
+                String errorCode = e.getErrorCode();
+                errorCode =
+                        errorCode.contains(RemoteFetchConfigurationConstants.ERROR_CODE_DELIMITER) ?
+                                errorCode : RemoteFetchConfigurationConstants.
+                                REMOTE_FETCH_CONFIGURATION_MANAGEMENT_PREFIX + errorCode;
+                errorResponse.setCode(errorCode);
+            }
+            errorResponse.setDescription(e.getMessage());
+            status = Response.Status.INTERNAL_SERVER_ERROR;
+        } else {
+            status = Response.Status.INTERNAL_SERVER_ERROR;
+        }
+        return new APIError(status, errorResponse);
+    }
+
+    /**
+     * From the error message and the data this method returns error builder.
+     *
+     * @param errorMsg error message.
+     * @param data     data.
+     * @return ErrorResponse.Builder
+     */
+    private ErrorResponse.Builder getErrorBuilder(RemoteFetchConfigurationConstants.ErrorMessage errorMsg,
```

Comment:
This can fit into one line.

Reply:
Sorry. The line length will be 121 characters.

Reply:
Ack